collection/lscq/asm_arm64.s

```diff
@@ -0,0 +1,89 @@
+#include "textflag.h"
+#include "funcdata.h"
+
+TEXT ·compareAndSwapUint128(SB), NOSPLIT, $0-41
+	MOVD	addr+0(FP), R0
+	MOVD	old1+8(FP), R2
+	MOVD	old2+16(FP), R3
+	MOVD	new1+24(FP), R4
+	MOVD	new2+32(FP), R5
+	MOVBU	·arm64HasAtomics+0(SB), R1
+    CBZ     R1, load_store_loop
+	ORR     R2, ZR, R6
```

Comment:
I'm personally not familiar with the aarch64 assembly, could you please explain more why we need these ORR instructions? Feel like MOVD is good enough, maybe I'm missing somethings.

BTW, I have another version of LSCQ arm64 support, but I don't have test environments at that time(about ~1 years ago), hope it can help :)

```
TEXT ·CompareAndSwapUint128(SB),NOSPLIT,$0
	MOVD  addr+0(FP), R7
	MOVD  old_0+8(FP), R2
	MOVD  old_1+16(FP), R3
	MOVD  R2, R4
	MOVD  R3, R5
	MOVD  new_0+24(FP), R8
	MOVD  new_1+32(FP), R9
	CASPD (R2, R3), (R7), (R8, R9)
	CMP   R2, R4
	CSET  EQ, R2
	CMP   R3, R5
	CSET  EQ, R3
	AND   R2, R3
	MOVD  R3, swapped+40(FP)
	RET

TEXT ·LoadUint128(SB),NOSPLIT,$0
	MOVD  addr+0(FP), R7
	MOVD  ZR, R2
	MOVD  ZR, R3
	MOVD  ZR, R8
	MOVD  ZR, R9
	CASPD (R2, R3), (R7), (R8, R9)
	MOVD  R2, val_0+8(FP)
	MOVD  R3, val_1+16(FP)
	RET
```

Reply:
Yes, `ORR R2, ZR, R6` behaves exactly the same as `MOVD R6, R2`.
The MOV register instruction is an alias of ORR instruction.
In [Arm A-profile A64 Instruction Set Architecture](https://documentation-service.arm.com/static/6245e8f0f7d10f7540e0c054):
> MOV \<Xd\>, \<Xm\>
> is equivalent to
> ORR \<Xd\>, XZR, \<Xm\>
> and is always the preferred disassembly.

Reply:
Hmm, could we use `MOVD` instead of `ORR`? I haven't seen any codes use `ORR <Xd>, XZR, <Xm>` to replace `MOV <Xd>, <Xm>` in the Go runtime for now. I feel like `MOVD` is more clear in this case.

Reply:
OK! I replace ORR with MOVD in the latest commit.